### DISCUSSION TITLE
Sites API - add endpoint to list all sites

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#1280] [Anything Else]
It's currently possible to create sites with Sites API, but there's no way to get a list of all sites from your Plausible account. Would be nice to add an endpoint for that as well.

(feedback via email)

**Comment by agraddy** [comment]
One other note is that there is not a basic GET call that can be made without knowing additional information. For example, every GET call requires knowing the SITE_ID but there is no way to get a SITE_ID from the API that I'm aware of.

I run a service where I would like to confirm that the user has entered their API key correctly but there is not a simple GET call I can make with just the API key.

**Reply by ukutaht** [reply]
What sort of API would you need for this apart from listing all sites? Look up a site by something other than `site_id`?

**Reply by agraddy** [reply]
Most APIs that I've worked with usually have an endpoint where you can call something like "/account" or "/me" and get basic account information. It would probably be nice to have an "/account" call that lists the plan, quota, current usage, and basic account info like name and email.

To give a better understanding of how I use the API, I run a site called [NumbersQ.com](https://www.numbersq.com) that allows users to track numbers from various APIs on one page. The steps to add an API number typically involve the same steps across APIs. For APIs that are key based, I first ask the user to enter their API key:

![Screenshot 2022-11-18 6 08 15 PM](https://user-images.githubusercontent.com/6887241/202822947-044d8f40-45bf-4af4-91fc-316780aa53dc.png)

With most APIs, I can make a call to an endpoint and verify that the user entered a valid API key. For example, with MailChimp, I make a call to "/lists" to confirm that the API key is valid:
https://github.com/dashboardq/numbersq/blob/main/app/services/connections/MailchimpConnectionService.php#L26-L34

For Plausible, I just have to assume the user entered a valid API key and move on:
https://github.com/dashboardq/numbersq/blob/main/app/services/connections/PlausibleConnectionService.php#L25

Where a "/sites" endpoint would be really helpful for me is on the next page. Right now, the user has to type in their Site ID:

![Screenshot 2022-11-18 6 09 44 PM](https://user-images.githubusercontent.com/6887241/202823616-0e9f6e27-e2d5-45c2-a0a6-c89340974afa.png)

Ideally, I could make a call to "/sites" and then show the user a drop down of their available sites.

So to sum it up, I need these two items (that could be solved by the same /sites endpoint):

1. A GET endpoint that can be called without any arguments to confirm the API key is valid.
2. A GET endpoint that lists the available site ids.

A nice to have for me would be an endpoint like "/account" that could give me account details, especially overall usage data in relation to account quota.

PS My decision to open source NumbersQ ( https://github.com/dashboardq/numbersq ) was heavily inspired by your work with Plausible so thanks for the inspiration!

**Reply by ukutaht** [reply]
Yeah, this makes sense. Thanks for the input @agraddy :)

We'll definitely consider adding something like this in the future. Can't promise anything at the moment, our internal roadmap is quite busy/full already

**Comment by siebird** [comment]
Having this endpoint would eliminate data entry error. Currently, a customer needs to type in their domain vs creating a dropdown list where they could choose the site:

![CleanShot 2023-01-25 at 07 36 17](https://user-images.githubusercontent.com/10938/214565469-95819ed8-9089-45f5-a05b-29922d4bae36.png)

**Comment by GioPan04** [comment]
Is there anything new on this?
